### PR TITLE
chore: remove market data requirement from swappers

### LINKED ISSUE #5086: Remove market prices from swappers
See here for spec
https://www.notion.so/shapeshift/Swapper-market-data-things-54be36040b1249d7a21934f0461f9633?pvs=4

### LINKED ISSUE #5098: move affiliateId and affiliateBps minimums into consumer (outside SDK)
### Overview

goal - not have market data/prices within 0x and 1inch swappers

context - affiliateBps minimum logic requires market data to determine this. move this logic outside the SDK such that it's a client concern.

### References and additional details

.

### Acceptance Criteria

.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Removes dependency on market data inside the swappers, in preparation for SDK.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #5098 
closes #5086 

## Risk

Low risk of broken donations in zrx and 1inch swappers.

## Testing

Check donations are still working for zrx and 1inch swappers.

### Engineering

☝️

### Operations

☝️

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5159** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5159" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5159?utm_source=stack-comment).

File: src/state/apis/swappers/swappersApi.ts [hunk 1]
```diff
@@ -50,25 +25,57 @@ export const swappersApi = createApi({
...
+        const sellAmountBeforeFeesCryptoPrecision = fromBaseUnit(
+          sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          sellAsset.precision,
         )
-        await dispatch(
-          marketApi.endpoints.findByAssetId.initiate(getTradeQuoteInput.buyAsset.assetId),
+        const sellAmountBeforeFeesUsd = bnOrZero(sellAmountBeforeFeesCryptoPrecision).times(
+          sellAssetUsdRate,
+        )
+        // We use the sell amount so we don't have to make 2 network requests, as the receive amount requires a quote
```
Comment (0xApotheosis) [inline src/state/apis/swappers/swappersApi.ts:54]: Nice. This logic works at a request level, but the UI will still show as donating when `isDonationAmountBelowMinimum` and the quote request contains no donation.

We'll want to hide the donation button in these cases, which matches the current behaviour.

**Review by 0xApotheosis — COMMENTED** [review]
Code looking solid, one functional comment.

File: src/state/apis/swappers/swappersApi.ts (hunk 1 above)
Comment (woodenfurniture) [reply src/state/apis/swappers/swappersApi.ts:54]: Not sure I follow, is the UI displaying 0 donations in these cases?

Comment (woodenfurniture) [reply src/state/apis/swappers/swappersApi.ts:54]: Might be resolved in following PR as I recall fixing this

**Review by 0xApotheosis — APPROVED** [review]
Approved noting the donations issue is fixed on the PR on top of this.

**Pull request merged**